.github/workflows/ecosystem-ci-trigger.yml

```diff
@@ -22,44 +46,123 @@ jobs:
+              const workflowFileName = 'preview-release.yml';
+              const { data: workflow } = await github.rest.actions.getWorkflow({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                workflow_id: workflowFileName,
+              });
+              const workflowId = workflow.id;
+
+              // Wait for 'Preview release' workflow to complete
+              const maxRetries = 60; // Wait up to 10 minutes
+              const delay = 10000; // 10 seconds
+              let completed = false;
+
+              for (let i = 0; i < maxRetries; i++) {
+                let page = 1;
+                let found = false;
+
+                do {
+                  const { data: runsData } = await github.rest.actions.listWorkflowRuns({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    workflow_id: workflowId,
+                    per_page: 100,
+                    page: page,
+                  });
+
+                  const runs = runsData.workflow_runs.filter(
+                    run => run.head_sha === prData.commit
+                  );
+
+                  if (runs.length > 0) {
+                    found = true;
+                    const latestRun = runs[0];
+                    if (latestRun.status === 'completed') {
+                      if (latestRun.conclusion === 'success') {
+                        console.log('Preview release workflow completed successfully.')
+                        completed = true;
+                        break;
+                      } else {
+                        throw new Error(`Workflow "Preview release" failed.`);
+                      }
+                    }
+                  }
+
+                  if (runsData.workflow_runs.length < 100) {
+                    break; // No more pages
+                  }
+                  page++;
+                } while (!found);
+
+                if (completed) {
+                  break;
+                }
+
+                // Wait
+                await new Promise(resolve => setTimeout(resolve, delay));
+              }
+
+              if (!completed) {
+                throw new Error(`Workflow "Preview release" did not complete in time.`);
+              }
+
+              // Remove the 'rocket' reaction
+              await github.rest.reactions.deleteForIssueComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                comment_id: 2448418810,
+                reaction_id: reaction.id,
+              });
+            }
+
             if (hasTriagePermission) {
               console.log('Allowed')
               await github.rest.reactions.createForIssueComment({
                 owner: context.repo.owner,
                 repo: context.repo.repo,
-                comment_id: context.payload.comment.id,
+                comment_id: 2448418810,
```

Comment:
hardcoded comment id wont work

Reply:
it is for debugging, will resolve it soon!